classes/log/Repository.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * @file classes/log/Repository.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Repository
+ *
+ * @ingroup log
+ *
+ * @brief Operations for retrieving and modifying EmailLogEntry objects.
+ */
+
+namespace PKP\log;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\submission\Submission;
+use Illuminate\Support\Facades\Mail;
+use PKP\core\Core;
+use PKP\facades\Locale;
+use PKP\log\core\EmailLogEventType;
+use PKP\mail\Mailable;
+use PKP\user\User;
+use Illuminate\Support\Facades\DB;
+
+class Repository
+{
+    private EmailLogEntry $model;
+
+    public function __construct(EmailLogEntry $model)
+    {
+        $this->model = $model;
+    }
+
+
+    /**
+     * Get the from or to data as a string
+     *
+     * @param array $addressees Expects Mailable::$to or Mailable::$from
+     */
+    private function getContactString(array $addressees): string
+    {
+        $contactStrings = [];
+        foreach ($addressees as $addressee) {
+            $contactStrings[] = isset($addressee['name'])
+                ? '"' . $addressee['name'] . '" <' . $addressee['address'] . '>'
+                : $addressee['address'];
+        }
+        return join(', ', $contactStrings);
+    }
+
+
+    /**
+     * Stores the correspondent user ids of the all recipient emails.
+     *
+     * @param EmailLogEntry $entry
+     */
+    private function insertLogUserIds(EmailLogEntry $entry)
+    {
+        $recipients = $entry->recipients;
+
+        // We can use a simple regex to get emails, since we don't want to validate it.
+        $pattern = '/(?<=\<)[^\>]*(?=\>)/';
+        preg_match_all($pattern, $recipients, $matches);
+        if (!isset($matches[0])) {
+            return;
+        }
+
+        foreach ($matches[0] as $emailAddress) {
+            $user = Repo::user()->getByEmail($emailAddress, true);
+            if ($user instanceof \PKP\user\User) {
+                // We use replace here to avoid inserting duplicated entries
+                // in table (sometimes the recipients can have the same email twice).
+                DB::table('email_log_users')->updateOrInsert(
+                    ['email_log_id' => $entry->id, 'user_id' => $user->getId()]
+                );
+            }
+        }
+    }
+
+    /**
+     * Delete all log entries for an object.
+     *
+     * @param int $assocType
```

Comment:
These two `@param` phpdocs add nothing that isn't already in the typehint; remove them. (Also elsewhere.)